### DISCUSSION TITLE
Priority is incorrect if there is not enough capacity until the due date

### DISCUSSION THREAD

**Discussion by kepjoe** [frePPLe/frepple#596] [Q&A]
Hi,

Frepple prioritizes demand by priority, this mean demads with the lowest priority would be planned first and become the latest date
and the element with the demads with higest priority would be planned at the end and become the nearest production date.
This is working well when there is enough capacitie until the due date.

In case there is not enough capacitie until the due date, the demads with the lowest priority would be planned first and the demads with higest priority would be planned after the due date. In this case, the production sequence is no longer maintained.
see the attached picture as illustration.

is there a way to ensure that higher priority demads would always get an earlier date than lower priority demads?

Thanks & BR,

Joel
![Frepple](https://github.com/frePPLe/frepple/assets/106798161/0946fed7-e6fb-4089-8f8e-1968a8602924)

**Comment by jdetaeye** [comment]
Your analysis is correct.

You can set the parameter "plan.administrativeleadtime" to 999 to force an ASAP / forward-scheduling plan (instead of the default backward scheduling from the due date). Each demand is then planned as early as possible, regardless of their due date.
In such a plan the higher priority demands will always be planned before the lower priority demands, as you desire.

**Reply by kepjoe** [reply]
Thank you for your prompt response. I just set the parameter "plan.administrativeleadtime" to 999 but still getting the same issue in the case where there is not enough capacitie before the due date. seem like also when forward-scheduling demads with the lowest priority are planned first so that the capacity before the due date is fill and the rest(with higest priority) can only be planned after  the due date.

If we have 2 demands with differents priority and only one capacity, the element with the lowest Priority would be planned and not the  demads with higest priority, regardless of whether we plan forward or backward..
I would have expected the opposite behavior in this case.

I am running frepple with the following parameter:
frepple.solver_mrp(allowsplits = True,administrativeleadtime = 999,plantype = 1,loglevel = 2,constraints = 15,iterationmax = 500,minimumdelay = 86000).solve()

is there anything/parameter i can adjust?

**Comment by jdetaeye** [comment]
>> If we have 2 demands with differents priority and only one capacity, the element with the lowest Priority would be planned and not the demands with higest priority, regardless of whether we plan forward or backward..

Correct. Frepple plans demand per demand, both in the forward and backward planning modes.
Demands are sorted by their priority (lowest number first: 1, 2, 3, ...) and by due date when the priority is the same.

>> I would have expected the opposite behavior in this case.

Maybe it's just a misunderstanding & confusion on the priority value?     Low priority numbers indicate a high priority demand that will be planned first.

**Reply by kepjoe** [reply]
i understand the usage of priority. In our case we have some elements that have to be produce before others.

refering to your suggestion  [here](https://github.com/frePPLe/frepple/discussions/504)
we just try to compute externally priorities 1,2,3, 4... on each demand, such that we can control very strictly the sequence followed by the planning algorithm.
And this is working well when we have enough capacity.
But the sequence is no more maintened when there is not enough capacity before the due date.
i am trying to find if there is a way to keep the sequence when there is not enough capacity?

**Reply by jdetaeye** [reply]
>In our case we have some elements that have to be produce before others.

Such relations should be defined in your bill of materials or in operation-dependencies.    

Relying on the processing order of the algorithm isn't a good way to enforce this.   There can be perfectly valid reasons to have an operation sequence that deviates from your demand priorities sequence;
An example:
demandA (due on day 5)   has steps:     operation A1 (1day)   -> operation A2 (1 days) -> operation A3 (1 day)
demandB  (due on day 10)  has steps:     operation B1 (1day)   -> operation B2 (10 days) -> operation B3 (1 day)
-> even when we plan demandA first, it's normal that operation B1 starts before the operation A1.  It just needs more time in the subsequent operation B2.
